src/mumble/GlobalShortcut_xdp.cpp

```diff
@@ -0,0 +1,200 @@
+    for (GlobalShortcut *shortcut : qmShortcuts) {
+        initialShortcuts.append({shortcut->objectName(), {
+            { QStringLiteral("description"), shortcut->name }
+        }});
+
+        Shortcut ourShortcut = { i, {uint(i)}, shortcut->qvDefault, false};
+        m_ids[i] = shortcut->objectName();
+        Global::get().s.qlShortcuts << ourShortcut;
+        ++i;
+    }
+
+    QDBusArgument arg;
+    arg << initialShortcuts;
+    QDBusPendingReply<QDBusObjectPath> reply = s_shortcutsInterface->CreateSession({
+        { QLatin1String("session_handle_token"), "Mumble" },
+        { QLatin1String("handle_token"), QLatin1String("mumble") },
+        { QLatin1String("shortcuts"), QVariant::fromValue(arg) },
+    });
+    reply.waitForFinished();
+    if (reply.isError()) {
+        qWarning() << "Couldn't get reply";
+        qWarning() << "Error:" << reply.error().message();
+    } else {
+        QDBusConnection::sessionBus().connect(QString(),
+                                            reply.value().path(),
+                                            QLatin1String("org.freedesktop.portal.Request"),
+                                            QLatin1String("Response"),
+                                            this,
+                                            SLOT(gotGlobalShortcutsCreateSessionResponse(uint,QVariantMap)));
+    }
+}
+
+void GlobalShortcutXdp::shortcutActivated(const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options)
+{
+    Q_UNUSED(session_handle);
+    Q_UNUSED(timestamp);
+    Q_UNUSED(options);
+
+    for (GlobalShortcut *shortcut : qmShortcuts) {
+        if (shortcut_id == shortcut->objectName()) {
+            shortcut->triggered(true, shortcut_id);
+            shortcut->down(shortcut_id);
+        }
+    }
+}
+
+void GlobalShortcutXdp::shortcutDeactivated(const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options)
+{
+    Q_UNUSED(session_handle);
+    Q_UNUSED(timestamp);
+    Q_UNUSED(options);
+
+    for (GlobalShortcut *shortcut : qmShortcuts) {
+        if (shortcut_id == shortcut->objectName()) {
+            shortcut->triggered(false, shortcut_id);
+        }
+    }
+}
+
+void GlobalShortcutXdp::shortcutsConfigured(const QDBusObjectPath &session_handle, const XdpShortcuts &shortcuts)
+{
+    Q_UNUSED(session_handle);
+    if (m_shortcuts != shortcuts) {
+        m_shortcuts = shortcuts;
+        emit shortcutsChanged();
+    }
+}
+
+void GlobalShortcutXdp::run() {
+	// 🤘🤪
+}
+
+bool GlobalShortcutXdp::canDisable() {
+	return false;
+}
+
+GlobalShortcutXdp::ButtonInfo GlobalShortcutXdp::buttonInfo(const QVariant &v) {
+	ButtonInfo info;
+	bool ok;
+	unsigned int key = v.toUInt(&ok);
+	if (!ok) {
+		return info;
+	}
+
+    info.device       = tr("Desktop");
+    info.devicePrefix = QString();
+    const QString id = m_ids[key];
```

Comment:
Wouldn't it be better to use the `value` function here? As far as I can tell, we have no (definitive) guarantee that `key` is indeed contained in `m_ids` and in such a case using the bracket operator will automatically insert a default-constructed entry, which I assume is not what we want here.